src/fluree/db/nameservice/memory.cljc

```diff
@@ -21,82 +22,66 @@
-    #?(:clj  (get @data-atom addr-path)
-       :cljs (or (get @data-atom addr-path)
-                 (and platform/BROWSER (.getItem js/localStorage addr-path))))))
+  [store address]
+  (store/read store (address-path address)))
 
 (defn push!
-  [data-atom {commit-address   :address
-              nameservice-iris :ns
-              :as              commit-data}]
+  [store {commit-address   :address
+          nameservice-iris :ns
+          :as              commit-data}]
   (go
     (let [my-ns-iri   (some #(when (re-matches #"^fluree:memory:(.+)" (:id %)) (:id %)) nameservice-iris)
           commit-path (address-path commit-address)
-          head-path   (address-path my-ns-iri)]
-      (swap! data-atom
-             (fn [state]
-               (let [commit (get state commit-path)]
-                 (when-not commit
+          head-path   (address-path my-ns-iri)
+
+          commit (store/read store commit-path)
+          _      (when-not commit
                    (throw (ex-info (str "Unable to locate commit in memory, cannot push!: " commit-address)
                                    {:status 500 :error :db/invalid-db})))
-                 (log/debug "pushing:" my-ns-iri "referencing commit:" commit-address)
-                 (let [commit (assoc commit "address" commit-address)]
-                   (assoc state head-path commit)))))
-      #?(:cljs (and platform/BROWSER (.setItem js/localStorage address-path commit-path)))
+          commit* (assoc commit "address" commit-address)]
+      (store/write store head-path commit*)
       commit-data)))
 
 (defn lookup
-  [data-atom ledger-alias opts]
-  (go #?(:clj
-         (if-let [head-commit (read-address data-atom ledger-alias)]
-           (-> head-commit (get "address"))
-           (throw (ex-info (str "Unable to lookup ledger address from conn: "
-                                ledger-alias)
-                           {:status 500 :error :db/missing-head})))
-
-         :cljs
-         (if platform/BROWSER
-           (if-let [head-commit (read-address data-atom ledger-alias)]
-             (memory-address head-commit)
-             (throw (ex-info (str "Unable to lookup ledger address from localStorage: "
-                                  ledger-alias)
-                             {:status 500 :error :db/missing-head})))
-           (throw (ex-info (str "Cannot lookup ledger address with memory connection: "
-                                ledger-alias)
-                           {:status 500 :error :db/invalid-ledger}))))))
+  [store ledger-alias opts]
+  (go (if-let [head-commit (read-address store ledger-alias)]
+        (-> head-commit (get "address"))
+        (throw (ex-info (str "Unable to lookup ledger address from conn: "
+                             ledger-alias)
+                        {:status 500 :error :db/missing-head})))))
 
 (defn ledger-list
-  [state-atom opts]
-  (go (-> @state-atom keys)))
+  [store opts]
+  (go (->> (keys @(:storage-atom store))
```

Comment:
this breaks the abstraction barrier. there should be a list method on storage that takes a prefix that this function can use. nothing should be peeking inside of storage implementations and looking at the storage atom.